### PR TITLE
Implement #on_many_times listener

### LINKED ISSUE #20: Trigger event x times and then remove the event
Thinking about adding a new listener that receives the event name and a second param that will be how many times that event will be triggered.
It will work like `#once` listener, but will loop before removing the event from the list.

### PR DESCRIPTION
**Pull request by rscardinho**
This PR implements the `#on_many_times` listener.

**How it works**
When adding a listener using `#on_many_times` you'll need to provide how many times you want that calback to be called, along with the callback itself.

You will emit this new listener was you do for any other listeners, but the registered callback will run only as many times as you provided.

_Example:_
If you want to run a callback only for the next two times, you can register the listener like this:

``` ruby
emitter.on_many_times :many_times, 2, &callback
```

The callback will get called only two times when emitting `:many_times` event.

This method accepts only positive Integer numbers, raising an `ArgumentError` for both non-integer and non-positive integer.

Closes #20

### CODE REVIEW

File: lib/emittr/events.rb [hunk 1]
```diff
@@ -59,6 +59,25 @@ def once_any(&block)
...
+        raise ArgumentError, "#{times} must be an integer" unless times.is_a? Integer
+        raise ArgumentError, "#{times} can't be negative" if times < 0
+
+        callback = ::Emittr::Callback.new(&block)
+
+        off_block = proc do |args|
+          callback.call(*args)
+          block_to_send = callback.wrapper || callback.callback
+
+          off(:on_many_times, &block_to_send) if times == 1
```
Comment (talyssonoc) [inline lib/emittr/events.rb]: Shouldn't we pass the event name too instead of using `:on_many_times`?

**Review by talyssonoc — CHANGES_REQUESTED** [review]

File: lib/emittr/events.rb (hunk 1 above)
Comment (rscardinho) [reply lib/emittr/events.rb]: yeah, we can pass it as well

File: lib/emittr/events.rb [hunk 2]
```diff
@@ -59,6 +59,25 @@ def once_any(&block)
...
+        raise ArgumentError, "#{times} must be an integer" unless times.is_a? Integer
+        raise ArgumentError, "#{times} can't be negative" if times < 0
+
+        callback = ::Emittr::Callback.new(&block)
+
+        off_block = proc do |args|
+          callback.call(*args)
+          block_to_send = callback.wrapper || callback.callback
+
+          off(event, &block_to_send) if times == 1
```
Comment (talyssonoc) [inline lib/emittr/events.rb]: What do you think about changing the order of this line and the next one to make it clearer when it should be `off`ed (and change the value to `0`)?

Comment (rscardinho) [reply lib/emittr/events.rb]: with that, the condition will need to check if `times` is zero instead of one. works for me 😄

**Pull request merged**